File: examples/go/iam_create_user.go
// Crea un usuario IAM.
// Uso: go run iam_create_user.go [nombre-usuario]
package main

import (
	"context"
	"fmt"
	"log"
	"os"

	"github.com/aws/aws-sdk-go-v2/aws"
	"github.com/aws/aws-sdk-go-v2/config"
	"github.com/aws/aws-sdk-go-v2/credentials"
	"github.com/aws/aws-sdk-go-v2/service/iam"
)

func main() {
	userName := "mi-usuario"
	if len(os.Args) > 1 {
		userName = os.Args[1]
	}

	ctx := context.Background()
	cfg, err := config.LoadDefaultConfig(ctx,
		config.WithRegion("us-east-1"),
		config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider("test", "test", "")),
	)
	if err != nil {
		log.Fatalf("Error cargando configuracion: %v", err)
	}
	client := iam.NewFromConfig(cfg, func(o *iam.Options) {
		o.BaseEndpoint = aws.String("http://localhost:4566")
	})

	result, err := client.CreateUser(ctx, &iam.CreateUserInput{UserName: aws.String(userName)})
	if err != nil {
		log.Fatalf("Error creando el usuario (¿ya existe %q?): %v", userName, err)
	}
	fmt.Println("Usuario IAM creado:", aws.ToString(result.User.UserName))
	fmt.Println("ARN:", aws.ToString(result.User.Arn))
	fmt.Println("Siguiente paso: go run iam_create_policy.go")
}

File: examples/node/iam-create-user.js
// Crea un usuario IAM.
// Uso: node iam-create-user.js [nombre-usuario]
const AWS = require('aws-sdk');

const iam = new AWS.IAM({
  endpoint: 'http://localhost:4566',
  region: 'us-east-1',
  credentials: { accessKeyId: 'test', secretAccessKey: 'test' },
});

const userName = process.argv[2] || 'mi-usuario';

async function main() {
  const result = await iam.createUser({ UserName: userName }).promise();
  console.log('Usuario IAM creado:', result.User.UserName);
  console.log('ARN:', result.User.Arn);
  console.log('Siguiente paso: node iam-create-policy.js');
}

main().catch((error) => {
  if (error.code === 'EntityAlreadyExists') {
    console.error(`El usuario "${userName}" ya existe.`);
  } else {
    console.error('Error creando el usuario:', error.message);
  }
  process.exit(1);
});
